### DISCUSSION TITLE
How to differentiate the Asynchronous and synchronous publish mode of FastDDS

### DISCUSSION THREAD

**Discussion by Syamkamalnadhc** [eProsima/Fast-DDS#5079] [Q&A]
I am trying to understand the behavior of Asynchronous and synchronous publish modes of FastDDS. 

Working version:
FastDDS 2.6.7(Have a separate build for FastDDS ), and ROS2 - Humble

Referred the below link to use the publish modes in ROS2 application
https://github.com/ros2/rmw_fastrtps/blob/master/README.md

I used Fast DDS HelloworldSharedMem example by changing the publish mode to synchronous and asynchronous 
![sync-async-fastdds](https://github.com/user-attachments/assets/8c73f26a-6826-4c67-b05a-7d78130501bc)
 
Observations:
1. DDS application behaves like publisher starts sending strings after the subscriber starts running.
2. ROS2 application behaves like publisher starts sending strings without checking the subscriber running or not.
3. The default value set in Fast DDS is SYNCHRONOUS.
4. I found that in FastDDS uses flow controller under asynchronous publish mode and not using under synchronous mode.

Questions:
 1. What is the difference in the expected behavior between synchronous and asynchronous modes
 2. Which transport example (UDP/TCP/SHM) can I use in Fast DDS to differentiate the asynchronous and synchronous publish mode behavior?
 3. What will be reason for ros2 and FadtDDS publish subscribe behavior change?

**Comment by JesusPoderoso** [comment]
Hi @Syamkamalnadhc, thanks for using Fast DDS

> 1. What is the difference in the expected behavior between synchronous and asynchronous modes

Please refer to the [publication mode kind description](https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/core/policy/eprosimaExtensions.html#publishmodeqospolicykind) in the documentation

> 2. Which transport example (UDP/TCP/SHM) can I use in Fast DDS to differentiate the asynchronous and synchronous publish mode behavior?

Any delivery mechanism can be used, they are agnostic to the publication mode.

> 3. What will be reason for ros2 and FadtDDS publish subscribe behavior change?

Thread management.
